cm/list_test.go

```diff
@@ -14,3 +18,304 @@ func TestListMethods(t *testing.T) {
 		t.Errorf("got (%s) != want (%s)", string(got), string(want))
 	}
 }
+
+type listTestItem struct {
```

Comment:
Is all this machinery necessary?

If `List[T]` implements `json.Marshaler` and `json.Unmarshaler`, then the test cases can have these lists inline.

Reply:
`listTestItem` is to test structs.

The machinery is necessary to generalize the `Slice()` method & carry the type information to `reflect.DeepEqual`. It's either this or unrolling into many smaller test functions per type.

While `List` implements `Marshaler/Unmarshaler`, accessing the underlying slice is thru typed `Slice() []T` ( not a common interface ). So the options are:
- make a generic wrapper ( what I did )
- cast it ( which requires knowing which type to cast to )

`Unmarshal` is where it is needed, and I used it in `Marshal` as well for consistency/less typing.